magento/inventory
# MSI-2592: Alternative for IsProductSalableInterface when working with a list of items

### Description (*)

Deprecated IsProductSalableInterface, introduced new interfaces AreProudctsSalableInterface, IsProductSalableResultInterface and AreProductsSalableResultInterface. Replace IsproductSalableInterface usage with new AreProductsSalableInterface.
Notice: AreProudctsSalableInterface implementation still uses IsProudctSalableInterface implementation.

### Fixed Issues (if relevant)

1. https://github.com/magento/inventory/issues/2592 Alternative for IsProductSalableInterface when working with a list of items

### Manual testing scenarios (*)

1. Create orders with all available product types and verify no errors occurred.
2. Export and import all available product types and verify no errors occurred.
3. Verify API calls, including GraphQl and verify no errors occurred during requests.

### Questions or comments


### Contribution checklist (*)
